File: python/benchmark.py
from __future__ import print_function
from traildb import TrailDB

def loading():
    traildb = TrailDB("/mnt/data/wikipedia-history-small.tdb")
    user_edits = 0
    ip_edits = 0
    empty_str = ""

    for uuid, trail in traildb.trails():
        for event in trail:
            if event.user != empty_str:
                user_edits += 1
            elif event.ip != empty_str:
                ip_edits += 1

    print("User edits: {}".format(user_edits))
    print("IP edits: {}".format(ip_edits))

loading()

File: javascript/benchmark.js
const TrailDB = require('traildb').TrailDB;

const tdb = new TrailDB({
  path: '/mnt/data/wikipedia-history-small.tdb'
});

var ipEdits = 0;
var userEdits = 0;

for (var trail of tdb.trails()) {
  for (var event of trail.events({ toMap: true })) {
    if (event.user) {
      userEdits++; 
    } else if(event.ip) {
      ipEdits++;
    }
  }
}

console.log('User edits:', userEdits);
console.log('IP edits:', ipEdits);